### ISSUE TITLE
[ERROR] Simulator is not available

### ISSUE THREAD

**Issue by DouglasHennrich** [appcelerator/ioslib#97]
```
[TRACE] [ioslib] Selected iOS Simulator: iPhone Xs
[TRACE] [ioslib]   UDID    = C1DA8BBA-CF17-4F7F-961D-E38060AD1617
[TRACE] [ioslib]   iOS     = 12.4
[TRACE] [ioslib] Autoselected Xcode: 10.3
[TRACE] [ioslib] Checking if the simulator /Applications/Xcode.app/Contents/Developer/Applications/Simulator.app/Contents/MacOS/Simulator is already running
[TRACE] [ioslib] Simulator is not running
[TRACE] [ioslib] Tailing iPhone Xs Simulator system log: /Users/douglashennrich/Library/Logs/CoreSimulator/C1DA8BBA-CF17-4F7F-961D-E38060AD1617/system.log
[ERROR] Simulator is not available
```

full `system.log`:
[system.log](https://github.com/appcelerator/ioslib/files/3624246/system.log)

**Comment by cb1kenobi** [comment]
Which Titanium SDK version are you using? I'm pretty sure this error was fixed in ioslib@1.7.12 or 1.7.13 (I don't recall which).

**Comment by DouglasHennrich** [comment]
I'm using `TiSDK 7.5.1.GA` and `appc - 7.1.1`

I start to facing this problem when I download and install xCode 11 GM besides my xCode 10... but I remove that new xCode and even reinstall my xCode 10 from app store. I completely delete all simulators and download them again but nothing seems to make it work again.

Already set my xcode selection to previous one with `sudo xcode-select -switch /Applications/Xcode.app/`

**Comment by cb1kenobi** [comment]
Titanium SDK 7.5.1.GA does not support Xcode 11. Titanium SDK 7.5.1 sees Xcode 11, but doesn't handle a few of the changes in Xcode 11. Selected Xcode 10 won't fix the problem. I suspect removing Xcode 11 won't restore things either. You would also have to delete the iOS 13 and watchOS 6 sims. Huge pain.

You need Titanium SDK 8.2.0 to support Xcode 11 and even then, there are issues which have fixes (#98) coming in 8.2.1.

Another possible option is to surgically update ioslib in the 7.5.1.GA install directory, but this is not supported and may not solve the problem.

**Comment by DouglasHennrich** [comment]
The problem is... I'm not using xcode 11, I'm using 10.3 as mentioned on log: `[TRACE] [ioslib] Autoselected Xcode: 10.3`

**Comment by DouglasHennrich** [comment]
Yea I remove all simulators from iOS 13, by the way, I delete all simulator and download all from iOS 12 back again

**Comment by DouglasHennrich** [comment]
```shell
== Device Types ==
iPhone 4s (com.apple.CoreSimulator.SimDeviceType.iPhone-4s)
iPhone 5 (com.apple.CoreSimulator.SimDeviceType.iPhone-5)
iPhone 5s (com.apple.CoreSimulator.SimDeviceType.iPhone-5s)
iPhone 6 Plus (com.apple.CoreSimulator.SimDeviceType.iPhone-6-Plus)
iPhone 6 (com.apple.CoreSimulator.SimDeviceType.iPhone-6)
iPhone 6s (com.apple.CoreSimulator.SimDeviceType.iPhone-6s)
iPhone 6s Plus (com.apple.CoreSimulator.SimDeviceType.iPhone-6s-Plus)
iPhone SE (com.apple.CoreSimulator.SimDeviceType.iPhone-SE)
iPhone 7 (com.apple.CoreSimulator.SimDeviceType.iPhone-7)
iPhone 7 Plus (com.apple.CoreSimulator.SimDeviceType.iPhone-7-Plus)
iPhone 8 (com.apple.CoreSimulator.SimDeviceType.iPhone-8)
iPhone 8 Plus (com.apple.CoreSimulator.SimDeviceType.iPhone-8-Plus)
iPhone X (com.apple.CoreSimulator.SimDeviceType.iPhone-X)
iPhone Xs (com.apple.CoreSimulator.SimDeviceType.iPhone-XS)
iPhone Xs Max (com.apple.CoreSimulator.SimDeviceType.iPhone-XS-Max)
iPhone Xʀ (com.apple.CoreSimulator.SimDeviceType.iPhone-XR)
iPad 2 (com.apple.CoreSimulator.SimDeviceType.iPad-2)
iPad Retina (com.apple.CoreSimulator.SimDeviceType.iPad-Retina)
iPad Air (com.apple.CoreSimulator.SimDeviceType.iPad-Air)
iPad mini 2 (com.apple.CoreSimulator.SimDeviceType.iPad-mini-2)
iPad mini 3 (com.apple.CoreSimulator.SimDeviceType.iPad-mini-3)
iPad mini 4 (com.apple.CoreSimulator.SimDeviceType.iPad-mini-4)
iPad Air 2 (com.apple.CoreSimulator.SimDeviceType.iPad-Air-2)
iPad Pro (9.7-inch) (com.apple.CoreSimulator.SimDeviceType.iPad-Pro--9-7-inch-)
iPad Pro (12.9-inch) (com.apple.CoreSimulator.SimDeviceType.iPad-Pro)
iPad (5th generation) (com.apple.CoreSimulator.SimDeviceType.iPad--5th-generation-)
iPad Pro (12.9-inch) (2nd generation) (com.apple.CoreSimulator.SimDeviceType.iPad-Pro--12-9-inch---2nd-generation-)
iPad Pro (10.5-inch) (com.apple.CoreSimulator.SimDeviceType.iPad-Pro--10-5-inch-)
iPad (6th generation) (com.apple.CoreSimulator.SimDeviceType.iPad--6th-generation-)
iPad Pro (11-inch) (com.apple.CoreSimulator.SimDeviceType.iPad-Pro--11-inch-)
iPad Pro (12.9-inch) (3rd generation) (com.apple.CoreSimulator.SimDeviceType.iPad-Pro--12-9-inch---3rd-generation-)
iPad mini (5th generation) (com.apple.CoreSimulator.SimDeviceType.iPad-mini--5th-generation-)
iPad Air (3rd generation) (com.apple.CoreSimulator.SimDeviceType.iPad-Air--3rd-generation-)
Apple TV (com.apple.CoreSimulator.SimDeviceType.Apple-TV-1080p)
Apple TV 4K (com.apple.CoreSimulator.SimDeviceType.Apple-TV-4K-4K)
Apple TV 4K (at 1080p) (com.apple.CoreSimulator.SimDeviceType.Apple-TV-4K-1080p)
Apple Watch - 38mm (com.apple.CoreSimulator.SimDeviceType.Apple-Watch-38mm)
Apple Watch - 42mm (com.apple.CoreSimulator.SimDeviceType.Apple-Watch-42mm)
Apple Watch Series 2 - 38mm (com.apple.CoreSimulator.SimDeviceType.Apple-Watch-Series-2-38mm)
Apple Watch Series 2 - 42mm (com.apple.CoreSimulator.SimDeviceType.Apple-Watch-Series-2-42mm)
Apple Watch Series 3 - 38mm (com.apple.CoreSimulator.SimDeviceType.Apple-Watch-Series-3-38mm)
Apple Watch Series 3 - 42mm (com.apple.CoreSimulator.SimDeviceType.Apple-Watch-Series-3-42mm)
Apple Watch Series 4 - 40mm (com.apple.CoreSimulator.SimDeviceType.Apple-Watch-Series-4-40mm)
Apple Watch Series 4 - 44mm (com.apple.CoreSimulator.SimDeviceType.Apple-Watch-Series-4-44mm)
== Runtimes ==
iOS 12.4 (12.4 - 16G73) - com.apple.CoreSimulator.SimRuntime.iOS-12-4 
tvOS 12.4 (12.4 - 16M567) - com.apple.CoreSimulator.SimRuntime.tvOS-12-4 
watchOS 5.3 (5.3 - 16U567) - com.apple.CoreSimulator.SimRuntime.watchOS-5-3 
== Devices ==
-- iOS 12.4 --
    iPhone Xs (C1DA8BBA-CF17-4F7F-961D-E38060AD1617) (Booted) 
-- tvOS 12.4 --
-- watchOS 5.3 --
== Device Pairs ==
```

**Comment by cb1kenobi** [comment]
Right, but simulator `C1DA8BBA-CF17-4F7F-961D-E38060AD1617` could be a `12.2` sim created by Xcode 11.

If you ever get stuck with your sims, you an reset them with this handy script I wrote: https://gist.github.com/cb1kenobi/19dc4c547864a96bfa9027aae5631e67.

**Comment by DouglasHennrich** [comment]
How can I completely uninstall `xcrun`, `xcode` and all it's files? Because I create the new simulator only after reinstall xcode 10... maybe my `xcrun` was changed by that previous xcode 11 installation

**Comment by DouglasHennrich** [comment]
> Right, but simulator `C1DA8BBA-CF17-4F7F-961D-E38060AD1617` could be a `12.2` sim created by Xcode 11.
> 
> If you ever get stuck with your sims, you an reset them with this handy script I wrote: https://gist.github.com/cb1kenobi/19dc4c547864a96bfa9027aae5631e67.

I run your code, it delete all simulators and re-create it but the problem remains =/

**Comment by DouglasHennrich** [comment]
```
xcrun --version
xcrun version 43.1
```

**Comment by cb1kenobi** [comment]
I don't think you should remove `xcrun`. FWIW, my machine says `xcrun version 48.`, but I have Xcode 9.4.1, 10.2.1, and 11.0 installed.

**Comment by cb1kenobi** [comment]
At the end of the day, Titanium is complaining that the simulator is not available because it's choosing one Xcode's simctl to detect a simulator it doesn't know anything about that another Xcode does know about. It's a huge mess and #98 fixes it.

**Comment by DouglasHennrich** [comment]
Any idea how can I fix this problem? Already tried to uninstall xcode and install it again from scratch

**Comment by cb1kenobi** [comment]
Either manually update ioslib in `~/Library/Application Support/Titanium/mobilesdk/osx/7.5.1.GA/node_modules/ioslib` to the version in #98 or completely remove Xcode 11 along with all of your simulators, `sudo xcode-select -s /path/to/Xcode10.app`, and run the script in the gist to recreate the sims.

**Comment by DouglasHennrich** [comment]
So i solve the problem uninstalling my xCode with CleanMyMac app, after that I have delete some folders on Library and reset my `xcode-select -r`. Now I have install the xCode from App Store again and now it's working 🎉

**Comment by cb1kenobi** [comment]
OK, cool!

**Comment by albert0m** [comment]
stuck in a similar context. Appstore updated automatically to Xcode 11 (disaster), too many bugs on my app, so I decided to switch back to Xcode 10.3 - downloaded the xip and reinstalled on different location. 
Since switching between Xcode versions didn't really work with Titanium, decided to remove everything. All versions of Xcode, simulators, nuked them all.
 Reinstalled only Xcode 10.3 but all of the sims were missing. I added them manually from the Devices/Simulators section, but still not working. Run your script, the simulators are listed but I always get the 

[ERROR] Simulator is not available

how can I update ioslib manually? I really need to publish an update :(

**Comment by albert0m** [comment]
ok, replacing ioslib fixed that error. 
The simulator boots but I get this other error:

```
[TRACE] [ioslib] Simulator is booted
[TRACE] [ioslib] iPhone 7 Plus iOS 12.4 Simulator started
[TRACE] [ioslib] Running: osascript "/Users/xxx/Library/Application Support/Titanium/mobilesdk/osx/7.5.1.v20181211045525/node_modules/ioslib/lib/sim_focus.scpt" "Simulator"
[TRACE] [ioslib] Detected crash file: /Users/xxx/Library/Logs/DiagnosticReports/xxx_2019-09-25-123817_iMac-27.crash
[ERROR] Detected crash:
[ERROR]   /Users/xxx/Library/Logs/DiagnosticReports/xxx_2019-09-25-123817_iMac-27.crash
[ERROR] Note: this crash may or may not be related to running your app.
```

the report says, more or less, 


```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
abort() called
```

**Comment by cb1kenobi** [comment]
@albert0m Cool, glad replacing ioslib solved the first problem. As for the crash log, I would just try again and possibly clear out any crash logs (who keeps these anyways?) and give it another go.

On a side note, I see you're using Titanium SDK 7.5.1. There are known issues if Xcode 11 is installed on your machine. SDK 7.5.1 sees Xcode 11, but chokes on the simulators... just by having Xcode 11 installed. If you have Xcode 11 installed, try moving the folder from the Applications directory to your desktop (temporarily so Titanium SDK 7.5.1 can't find it), then make sure Xcode 10 is selected via xcode-select.

**Comment by albert0m** [comment]
Xcode had been uninstalled completely, using AppCleaner. The only few things that I didn't manage to uninstall were the cli-tools (like xcode-select and such).
I'm using 7.5.1 because the app still doesn't run with 8.x

Xcode has been set to 10.3
```
xcode-select -p
/Applications/Xcode10.3.app/Contents/Developer
```

clean build, got rid of the crash log, but still:

```
[TRACE] [ioslib] Running: osascript "/Users/xxx/Library/Application Support/Titanium/mobilesdk/osx/7.5.1.v20181211045525/node_modules/ioslib/lib/sim_focus.scpt" "Simulator"
[TRACE] [ioslib] Exited with code: 0
```

other things I could do?

**Comment by cb1kenobi** [comment]
Not sure what the problem is. The `Exited with code: 0` is a good thing. If the exit code is greater than zero, then that implies a failure.

**Comment by albert0m** [comment]
it was indeed a good thing, after a reboot this morning everything seems to work as normal. Thank you for your help.
